src/core/configuration/Colors.ts

```diff
@@ -380,29 +457,22 @@ export class ColorAllocator {
     return color;
   }
 
-  assignTeamColor(team: Team): Colord {
-    switch (team) {
-      case ColoredTeams.Blue:
-        return blue;
-      case ColoredTeams.Red:
-        return red;
-      case ColoredTeams.Teal:
-        return teal;
-      case ColoredTeams.Purple:
-        return purple;
-      case ColoredTeams.Yellow:
-        return yellow;
-      case ColoredTeams.Orange:
-        return orange;
-      case ColoredTeams.Green:
-        return green;
-      case ColoredTeams.Bot:
-        return botColor;
-      default:
-        return this.availableColors[
-          simpleHash(team) % this.availableColors.length
-        ];
+  assignTeamColor(team: Team, playerId?: string): Colord {
+    const teamColors = this.getTeamColorVariations(team);
```

Comment:
I think you removed the cache lookup by mistake:
```suggestion
    // If we have a player ID, assign a consistent color variation to this player
    if (playerId) {
      if (this.teamPlayerColors.has(playerId)) {
        return this.teamPlayerColors.get(playerId)!;
      }
    }

    const teamColors = this.getTeamColorVariations(team);
```